test/jdk/java/awt/image/multiresolution/MultiDisplayTest.java

```diff
@@ -23,61 +23,104 @@
 
 
 /*
-  @test
-  @bug 8142861 8143062 8147016
-  @summary Check if multiresolution image behaves properly
-           on HiDPI + non-HiDPI display pair.
-  @author a.stepanov
-  @library /test/lib
-  @build jdk.test.lib.Platform
-  @run applet/manual=yesno MultiDisplayTest.html
-*/
-
-
-import java.applet.Applet;
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.image.*;
+ * @test
+ * @bug 8142861 8143062 8147016
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @summary Check if multiresolution image behaves properly
+ *         on HiDPI + non-HiDPI display pair.
+ * @run main MultiDisplayTest
+ */
+
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Dialog;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.BaseMultiResolutionImage;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JFrame;
 
 import jdk.test.lib.Platform;
 
-public class MultiDisplayTest extends Applet {
+public class MultiDisplayTest {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            """
+             This test is for OS X or Windows only.
+             For other OSes please simply push "Pass".
```

Comment:
`please simply push "Pass"` usually means that it can be replaced with `PassFailJFrame.forcePass()`, but in this case the `@requires (os.family...` can be used to avoid running the test on unwanted platforms.

Reply:
Added jtreg `@requires` tag.